server/src/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -278,6 +278,8 @@
     @Inject
     private EndPointSelector selector;
 
+    protected final ConfigKey<Integer> TemplatePreloaderPoolSize = new ConfigKey<Integer>("Advanced", Integer.class, "template.preloader.pool.size", "8", "Size of the TemplateManager threadpool", false);
```

Comment:
@SudharmaJain You have to implement the Configurable interface. Check for its usage in code.